### PR TITLE
x-cache

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -9,6 +11,7 @@ const memo = async(key, fn, ttl, forceUpdate) => {
     }
   }
   const result = await fn();
+  timestamps[key] = new Date().getTime();
```
Comment (jgallen23) [inline index.js]: could you just set the header here? and then not have to worry about the timetamps object

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: index.js [hunk 2]
```diff
@@ -9,6 +9,9 @@ const memo = async(key, fn, ttl, forceUpdate) => {
     }
   }
   const result = await fn();
+  if (result.headers) {
```
Comment (jgallen23) [inline index.js]: what is headers _isn't_ set?

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: index.js [hunk 3]
```diff
@@ -9,6 +9,12 @@ const memo = async(key, fn, ttl, forceUpdate) => {
     }
   }
   const result = await fn();
+  // always set the header showing when it was cached:
+  if (!result.headers) {
+    result.headers = { 'x-cache': new Date().getTime() };
```
Comment (jgallen23) [inline index.js]: x-arc-cache please

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (jgallen23) [reply index.js]: there's dupe code here.  can you just do 
```
if (!results.headers) {
  results.headers = {}
}
results.headers[...]
```

Comment (orthagonal) [reply index.js]: That will break if results is not an object.

Comment (orthagonal) [reply index.js]: For example if result is a number, adding result.headers to it does nothing and the next line will throw ' Cannot set property 'x-arc-cache' of undefined' when you try to set the header.

Comment (jgallen23) [reply index.js]: how is that different than what you have?

**Pull request merged**